tests/unit/core/middleware/test_trailingSlashMiddleware.js

```diff
@@ -182,6 +182,59 @@ describe(__filename, () => {
     sinon.assert.notCalled(fakeNext);
   });
 
+  it('should trim params', () => {
```

Comment:
I think this needs a couple of additional tests to cover a couple of additional combinations:

* Make sure paths excluded from needing trailing slashes and trimming works.
* Make sure trimming works when there's no trailing slash and it's not excluded.
* A test to cover re-encoding.